zipline/testing/fixtures.py

```diff
@@ -292,3 +299,126 @@ def init_class_fixtures(cls):
         start_loc = end_loc - cls.TRADING_DAY_COUNT
 
         cls.trading_days = all_days[start_loc:end_loc + 1]
+
+
+class WithPipelineEventDataLoader(WithAssetFinder):
+    """
+    ZiplineTestCase mixin providing common test methods/behaviors for event
+    data loaders.
+
+    `get_sids` must return the sids being tested.
+    `get_dataset` must return {sid -> pd.DataFrame}
+    `loader_type` must return the loader class to use for loading the dataset
+    `make_asset_finder` returns a default asset finder which can be overridden.
+    """
+    @abc.abstractproperty
+    def get_sids(cls):
+        raise NotImplementedError('get_sids')
+
+    @abc.abstractproperty
+    def get_dataset(self):
+        raise NotImplementedError('get_dataset')
+
+    @abc.abstractproperty
+    def loader_type(self):
+        raise NotImplementedError('loader_type')
+
+    @classmethod
+    def make_asset_finder(cls):
```

Comment:
the way to implement this pattern would be to override the class method `make_equities_info` and have the return value of `make_simple_equity_info(...)`; however, I am not sure if it should be this fixture's job to setup the equities info.